std/hl/_std/StringBuf.hx

```diff
@@ -101,6 +101,10 @@
 			throw "Invalid unicode char " + c;
 	}
 
+	public function clear():Void {
+		pos = 0;
```

Comment:
I wonder if the `hl.Bytes` instance should also be reinitialized for memory reasons. In theory you could have a huge StringBuf that never releases its memory after being cleared.

Reply:
The `StringBuf` staying huge after being cleared is intentional! For example, if you built a huge string once:
```haxe
hl.Gc.major();
Sys.println("before start: " + hl.Gc.stats());

var sb = new StringBuf();
for (i in 0...200000) {
	sb.add("hello");
}

hl.Gc.major();
Sys.println("after filling: " + hl.Gc.stats());
```
```
before start: {currentMemory : 589824, allocationCount : 137, totalAllocated : 7104}
after filling: {currentMemory : 3145728, allocationCount : 191, totalAllocated : 7706696}
```
You may also want to build another one in the future. In that case, you don't pay the cost of allocating buffers. Multiple, if dealing with small elements and the internals need to constantly reallocate and blit.
```haxe
@:privateAccess sb.pos = 0; // clear()
for (i in 0...200000) {
	sb.add("world");
}

hl.Gc.major();
Sys.println("after reusing: " + hl.Gc.stats());
```
```
after reusing: {currentMemory : 3145728, allocationCount : 212, totalAllocated : 7707912}
```

If you worry however about `hl.Bytes` leaking, the semantics around it did not change - if `Bytes` are indeed GC-able, they should leak if and only if the `StringBuf` also leaks (user error 😅), the same as it currently functions. (Ditto for compacting the heap/releasing memory back to the OS - there's currently nothing stopping users from creating big, non-clearable StringBufs. If this was a concern, this would be a concern right now on stable.)

Reply:
I see what you're saying, it's just that this is not exactly the behavior I'd expect when clearing a buffer. From a design perspective, I expect `existingBuf = new StringBuf()` and `existingBuf.clear()` to have the same memory behavior. Introducing a subtle difference like this can easily lead to confusion.